pkg/host/internal/sriov/sriov.go

```diff
@@ -221,44 +223,41 @@ func (s *sriov) DiscoverSriovDevices(storeManager store.ManagerInterface) ([]sri
 			continue
 		}
 
+		if !vars.DevMode {
+			if !sriovnetworkv1.IsSupportedModel(device.Vendor.ID, device.Product.ID) {
+				log.Log.Info("DiscoverSriovDevices(): unsupported device", "device", device)
+				continue
+			}
+		}
+
 		driver, err := s.dputilsLib.GetDriverName(device.Address)
 		if err != nil {
 			log.Log.Error(err, "DiscoverSriovDevices(): unable to parse device driver for device, skipping", "device", device)
 			continue
 		}
 
-		deviceNames, err := s.dputilsLib.GetNetNames(device.Address)
-		if err != nil {
-			log.Log.Error(err, "DiscoverSriovDevices(): unable to get device names for device, skipping", "device", device)
-			continue
-		}
+		pfNetName := s.networkHelper.TryGetInterfaceName(device.Address)
 
-		if len(deviceNames) == 0 {
-			// no network devices found, skipping device
+		if pfNetName == "" {
+			log.Log.Error(err, "DiscoverSriovDevices(): unable to get device names for device, skipping", "device", device.Address)
 			continue
 		}
 
-		if !vars.DevMode {
-			if !sriovnetworkv1.IsSupportedModel(device.Vendor.ID, device.Product.ID) {
-				log.Log.Info("DiscoverSriovDevices(): unsupported device", "device", device)
-				continue
-			}
+		link, err := s.netlinkLib.LinkByName(pfNetName)
+		if err != nil {
+			log.Log.Error(err, "DiscoverSriovDevices(): unable to get Link for device, skipping", "device", device.Address)
+			continue
 		}
 
 		iface := sriovnetworkv1.InterfaceExt{
 			PciAddress: device.Address,
 			Driver:     driver,
 			Vendor:     device.Vendor.ID,
 			DeviceID:   device.Product.ID,
-		}
-		if mtu := s.networkHelper.GetNetdevMTU(device.Address); mtu > 0 {
-			iface.Mtu = mtu
-		}
-		if name := s.networkHelper.TryGetInterfaceName(device.Address); name != "" {
-			iface.Name = name
-			iface.Mac = s.networkHelper.GetNetDevMac(name)
-			iface.LinkSpeed = s.networkHelper.GetNetDevLinkSpeed(name)
-			iface.LinkType = s.GetLinkType(name)
+			Mtu:        link.Attrs().MTU,
+			Mac:        string(link.Attrs().HardwareAddr),
```

Comment:
use `link.Attrs().HardwareAddr,String()` instead

Reply:
IDE autocompleted it :(. yes i will fix. which probably related to CI failure.